### PR TITLE
Resolved adding cards in unopened deck issue

### PR DESCRIPTION
**Pull request by haveaqiupill**

### CODE REVIEW

File: src/main/java/cardibuddy/logic/commands/SearchCardCommand.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import cardibuddy.model.flashcard.SearchCardKeywordsPredicate;
+
+
+/**
+ * Finds and lists all decks and cards in cardibuddy whose title contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class SearchCardCommand extends Command {
+
+    public static final String COMMAND_WORD = "searchcard";
```
Comment (airshhlay) [inline src/main/java/cardibuddy/logic/commands/SearchCardCommand.java]: do we want our command word to be like this though? cos it's a bit inconsistent
why not search card?

**Review by airshhlay — COMMENTED** [review]
Maybe there's a better way to phrase our commands?
We can start with `search card` and `search deck` and within the Search command parser itself, check to see if it is a card or deck before returning the SearchCard or SearchDeck commands.

Just a suggestion! You may have done this because of certain restrictions in your implementation.

File: src/main/java/cardibuddy/logic/commands/SearchCardCommand.java (hunk 1 above)
Comment (haveaqiupill) [reply src/main/java/cardibuddy/logic/commands/SearchCardCommand.java]: changed to "search card" instead!

**Comment by sigmund-c** [comment]
It shows up this error message whenever I try to add a flashcard, though the flashcard does get added

`Caused by: java.lang.UnsupportedOperationException
	at java.base/java.util.Collections$UnmodifiableCollection.add(Collections.java:1058)
	at cardibuddy.logic.commands.AddCommand.execute(AddCommand.java:114)
	at cardibuddy.logic.LogicManager.execute(LogicManager.java:49)
	at cardibuddy.ui.MainWindow.executeCommand(MainWindow.java:254)
	at cardibuddy.ui.CommandBox.handleCommandEntered(CommandBox.java:38)
	... 59 more`

**Pull request merged**